vendor/github.com/containers/image/v5/signature/fulcio_cert.go

```diff
@@ -174,10 +176,17 @@ func (f *fulcioTrustRoot) verifyFulcioCertificateAtTime(relevantTime time.Time,
 	}
 
 	// == Validate the OIDC subject
-	if !slices.Contains(untrustedCertificate.EmailAddresses, f.subjectEmail) {
-		return nil, internal.NewInvalidSignatureError(fmt.Sprintf("Required email %s not found (got %#v)",
-			f.subjectEmail,
-			untrustedCertificate.EmailAddresses))
+	if !slices.Contains(untrustedCertificate.EmailAddresses, f.subjectEmail) && !strings.Contains(untrustedCertificate.URIs[0].String(), f.URI) {
+		if len(untrustedCertificate.EmailAddresses) > 0 {
+			return nil, internal.NewInvalidSignatureError(fmt.Sprintf("Required email %s not found (got %#v)",
+				f.subjectEmail,
+				untrustedCertificate.EmailAddresses))
+		}
+		if len(untrustedCertificate.URIs) > 0 {
+			return nil, internal.NewInvalidSignatureError(fmt.Sprintf("Required URI %s not found (got %#v)",
+				f.URI,
+				untrustedCertificate.URIs))
+		}
```

Comment:
- ⚠️ If the policy requires _both_ an email address and an URI, this must not accept signatures which only have one of them
- This is invoking `slices.Contains(…, "")` if the field is not set. That’s at the very least unclean.
- I suspect the error reporting logic should also be tightened. It must primarily decide based on the _policy_, not on attacker-set `untrusted*` fields.

Reply:
> If the policy requires both an email address and an URI, this must not accept signatures which only have one of them

I believe this would be an invalid policy with cosign. One or the other.

Reply:
Then such an invalid policy must be rejected at parsing/initialization time, not processed in an unexpected way.